--- CMakeLists.txt	2024-01-17 21:00:57.594529869 +0100
+++ /tmp/CMakeLists.txt	2024-01-17 21:00:53.546652901 +0100
@@ -21,7 +21,7 @@
         PREFIX=${CMAKE_INSTALL_PREFIX}
         LIBDIR=${CMAKE_INSTALL_FULL_LIBDIR}
         INCLUDEDIR=${CMAKE_INSTALL_FULL_INCLUDEDIR}
-        ${CARGO} build --release --no-default-features --features jsonrpc
+        CARGO_TARGET_DIR=${CLICK_BUILD_DIR} CARGO_HOME=~/.clickable/cargo ${CARGO} build --target=${CLICK_ARCH_RUST} --release --no-default-features --features jsonrpc
 
 	# Build in `deltachat-ffi` directory instead of using
 	# `--package deltachat_ffi` to avoid feature resolver version
@@ -45,6 +45,6 @@
 )
 
 install(FILES "deltachat-ffi/deltachat.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-install(FILES "target/release/libdeltachat.a" DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES "target/release/libdeltachat.${DYNAMIC_EXT}" DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES "target/release/pkgconfig/deltachat.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+install(FILES "${CLICK_BUILD_DIR}/${CLICK_ARCH_RUST}/release/libdeltachat.a" DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES "${CLICK_BUILD_DIR}/${CLICK_ARCH_RUST}/release/libdeltachat.${DYNAMIC_EXT}" DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES "${CLICK_BUILD_DIR}/${CLICK_ARCH_RUST}/release/pkgconfig/deltachat.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
